public_profiler/v3.5-DD/ac.js

```diff
@@ -67,32 +67,23 @@ function ACChangeArmor()
     return;
 
   sheet().ACArmor.value  = 0;
-  if ( sheet().Armor1Worn.checked )
-  {
-      sheet().Armor1Carried.checked = true;
-      sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor1Bonus)));
-  }
-
-  if ( sheet().Armor3Worn.checked )
-  {
-      sheet().Armor3Carried.checked = true;
-      sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor3Bonus)));
-  }
 
-  if ( sheet().Armor4Worn.checked )
-  {
-      sheet().Armor4Carried.checked = true;
-          sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value, GetNum(sheet().Armor4Bonus)));
+  for (var i = 1; i <= 4; i++) {
+      if (!sheet()["Armor" + i + "Worn"].checked) {
+          sheet()["Armor" + i + "Carried"].checked = true;
+          sheet().ACArmor.value = Clean(Add(sheet().ACArmor.value,
+                                            GetNum(sheet()["Armor" + i + "Bonus"])));
+      }
   }
 
   sheet().ACShield.value  = 0;
   if ( sheet().Armor2Worn.checked )
   {
       sheet().Armor2Carried.checked = true;
-     sheet().ACShield.value = GetNum(sheet().Armor2Bonus);
+      sheet().ACShield.value = GetNum(sheet().Armor2Bonus);
```

Comment:
Oh, look - the above is wrongly covering i = 2; it needs an adapted version to handle that case differently.